File: apiendpoint/app/models/oqc.py
from sqlalchemy import Column, Integer, String, Text, TIMESTAMP, DECIMAL, ForeignKey
from sqlalchemy.orm import relationship
from app.database.base import Base

class OQC(Base):
    __tablename__ = "oqc"
    id = Column(Integer, primary_key=True, index=True)
    part_number = Column(String(100), ForeignKey("master_prod.part_number"))
    lot_number = Column(String(100))
    quantity_good = Column(DECIMAL(10, 2), nullable=False)  # Sesuai database
    quantity_ng = Column(DECIMAL(10, 2), default=0.00)
    inspection_date = Column(TIMESTAMP)
    inspector_id = Column(Integer, ForeignKey("users.id"))  # Sesuai database
    notes = Column(Text)
    created_at = Column(TIMESTAMP)

File: apiendpoint/node/models/oqc.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const OQC = sequelize.define('OQC', {
    id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
    part_number: { type: DataTypes.STRING(100), allowNull: false },
    lot_number: { type: DataTypes.STRING(255), allowNull: false, unique: true },
    quantity_good: { type: DataTypes.DECIMAL(10, 2), allowNull: false },
    quantity_ng: { type: DataTypes.DECIMAL(10, 0) },
    inspection_date: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
    inspector_id: { type: DataTypes.INTEGER },
    notes: { type: DataTypes.TEXT },
    created_at: { type: DataTypes.DATE }
}, { tableName: 'oqc', timestamps: false });

module.exports = OQC;
